### PR TITLE
stricter BTF platform processing

### PR DESCRIPTION
**Pull request by brycekahle**
### What does this PR do?

- Uses BTF platform names that match `btfhub-archive`, which come from `/etc/os-release` values. (`amazon` -> `amzn`, `oracle` -> `ol`, `redhat` -> `rhel`)
- Uses `x86_64` rather than `amd64` for architecture naming, to match btfhub-archive
- Enforces BTF platform names during BTF search/load, rejecting unknown values.
- Uses a python invoke task for the `btfhub-archive` processing, rather than a brittle bash script.
- Adds a CI variable for the branch to use from `btfhub-archive`, this will allow testing of changes to the btfhub-archive structure.
- Uses the CI variable in the BTF S3 upload path, so individual branches will not influence others.
- Removes unnecessary upload to `S3_PERMANENT_ARTIFACTS_URI`
- Add `KUBERNETES_CPU_REQUEST` to the `build_processed_btfhub_archive` Gitlab job, at the request of CI Reliability.

### Motivation

See above

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/ebpf/btf.go [hunk 1]
```diff
@@ -174,33 +214,33 @@ func (b *orderedBTFLoader) embeddedPath() (string, error) {
 
 var kernelVersionPatterns = []struct {
```
Comment (spencergilbert) [inline pkg/ebpf/btf.go:215]: **❓ question:** ‏This doesn't need SUSE/SLES?

**Review by spencergilbert — APPROVED** [review]
I had a couple of questions, but I didn't see anything blocking.

File: pkg/ebpf/btf.go [hunk 2]
```diff
@@ -31,6 +31,46 @@ import (
...
+
+func (p btfPlatform) String() string {
+	return string(p)
+}
+
+func btfPlatformFromString(platform string) (btfPlatform, error) {
+	switch platform {
+	case "amzn", "amazon":
+		return platformAmazon, nil
+	case "suse", "sles": //opensuse treated differently on purpose
```
[the remark this answers is not in the document]
Comment (brycekahle) [reply pkg/ebpf/btf.go:55]: ah, we are NOT including `opensuse` because the kernels are not equivalent to `sles`.

File: pkg/ebpf/btf.go (hunk 1 above)
Comment (brycekahle) [reply pkg/ebpf/btf.go:215]: Nope, these are patterns which let us infer possible platforms from the just the kernel version. SUSE/SLES doesn't have any identifying strings unfortunately.

Comment (spencergilbert) [reply pkg/ebpf/btf.go:215]: Cool - thanks!

**Review by iglendd — APPROVED** [review]

**Review by gjulianm — APPROVED** [review]

File: pkg/ebpf/btf.go [hunk 3]
```diff
@@ -174,33 +214,33 @@ func (b *orderedBTFLoader) embeddedPath() (string, error) {
...
-	{regexp.MustCompile(`\.el7\.`), []string{"redhat", "centos", "oracle"}},
-	{regexp.MustCompile(`\.el8(_\d)?\.`), []string{"redhat", "centos", "oracle"}},
-	{regexp.MustCompile(`\.el[7-8]uek\.`), []string{"oracle"}},
-	{regexp.MustCompile(`\.deb10\.`), []string{"debian"}},
-	{regexp.MustCompile(`\.fc\d{2}\.`), []string{"fedora"}},
+	{regexp.MustCompile(`\.amzn[1-2]\.`), []btfPlatform{platformAmazon}},
+	{regexp.MustCompile(`\.el7\.`), []btfPlatform{platformRedhat, platformCentOS, platformOracle}},
+	{regexp.MustCompile(`\.el8(_\d)?\.`), []btfPlatform{platformRedhat, platformCentOS, platformOracle}},
+	{regexp.MustCompile(`\.el[7-8]uek\.`), []btfPlatform{platformOracle}},
+	{regexp.MustCompile(`\.deb10\.`), []btfPlatform{platformDebian}},
```
Comment (nplanel) [inline pkg/ebpf/btf.go:223]: why only deb10 ?

Comment (brycekahle) [reply pkg/ebpf/btf.go:223]: deb11 already has BTF, and debian 9 does not follow this pattern.

**Pull request merged**